compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -2001,54 +2054,55 @@ TR::Register *OMR::X86::TreeEvaluator::integerMulEvaluator(TR::Node *node, TR::C
          // are evaluated by the generic analyser call below
          if (targetRegister == 0 && IS_32BIT_SIGNED(value)) // decomposition failed
             {
-            TR_X86OpCodes opCode;
-            if (firstChild->getReferenceCount() > 1 ||
-                firstChild->getRegister() != 0)
+            if(isByteOperation)
                {
-               if (value >= -128 && value <= 127)
+               // Set dependecy on EAX
+               TR::RegisterDependencyConditions  *multDependencies = generateRegisterDependencyConditions((uint8_t)1, 1, cg);
+               multDependencies->addPreCondition(targetRegister, TR::RealRegister::eax, cg);
```

Comment:
is it really the tagetRegister that should be in the pre condition - this seems wrong - you would want the incoming value in the pre and the target in the post no?

Reply:
@andrewcraik Can you please looks at the reviewed code.

Reply:
makes sense, I'd just move the intClobberEvaluate above the dependencies so that people know what target register is before they see the dependency conditions.

Reply:
@andrewcraik .. I have done the suggested changes.